paprika/analysis.py

```diff
@@ -275,42 +285,37 @@ def _run_mbar(self, prepared_data, verbose=False):
         # coordinates in each window's potential function
         u_kln = np.zeros([num_win, num_win, max_data_points], np.float64)
 
+        # Transpose force_constants and targets into "per window" format, instead of
+        # the "per restraint" format.
+        force_constants_T = np.asarray(force_constants).T
+        targets_T = np.asarray(targets).T
+
         # Note, the organization of k = coordinate windows, l = potential windows
         # seems to be opposite of the documentation. But I got wrong numbers the other way around.
         for k in range(num_win):  # Coordinate windows
             for l in range(num_win):  # Potential Windows
-                force_constants_T = np.asarray(force_constants).T[l, :, None]
-                targets_T = np.asarray(targets).T[l, :, None]
 
                 for r, rest in enumerate(active_rest):  # Restraints
-
                     # If this is a dihedral, we need to shift around restraint value
                     # on the periodic axis to make sure the lowest potential is used.
                     if rest.mask3 is not None and rest.mask4 is not None:
-                        target = np.asarray(targets).T[l][r]  # Taken from potential window, l
+                        target = targets_T[l, r]
                         bool_list = ordered_values[k][r] < target - 180.0  # Coords from coord window, k
                         ordered_values[k][r][bool_list] += 360.0
                         bool_list = ordered_values[k][r] > target + 180.0
                         ordered_values[k][r][bool_list] -= 360.0
 
                 # Compute the potential ... for each frame, sum the contributions for each restraint
                 # Note, we multiply by beta, and do some extra [l,:,None] to get the math operation correct.
-
                 u_kln[k, l, 0:N_k[k]] = np.sum(
-                    self.beta * force_constants_T * (ordered_values[k] - targets_T)**2, axis=0)
-
-        # Setup mbar calc, and get matrix of free energies, uncertainties
-        mbar = pymbar.MBAR(u_kln, N_k, verbose=verbose)
-        Deltaf_ij, dDeltaf_ij, Theta_ij = mbar.getFreeEnergyDifferences(compute_uncertainty=True)
+                    self.beta * force_constants_T[l, :, None] * (ordered_values[k] - targets_T[l, :, None])**2, axis=0)
 
+        g_k = np.ones([num_win], np.float64)
         # Should I subsample based on the restraint coordinate values? Here I'm
         # doing it on the potential.  Should be pretty close ....
-        if 'mbar-block' in self.methods:
+        if method == 'mbar-block':
```

Comment:
If blocking is requested, we calculate the statistical inefficiency from blocking, otherwise, the statistical inefficiency is 1 (i.e., there is no correlation among data points).

Reply:
(`g_k`)